lighthouse-core/audits/apple-touch-icon.js

```diff
@@ -51,18 +51,13 @@ class IosPwaIcon extends Audit {
    */
   static audit(artifacts) {
     let explanation;
-    const appleTouchIcons = artifacts.LinkElements.filter(
-      el => {
-        if (el.rel !== 'apple-touch-icon' && el.rel !== 'apple-touch-icon-precomposed'
-          || !el.href) {
+    const appleTouchIcons = artifacts.LinkElements
+      .filter(el => el.rel === 'apple-touch-icon' || el.rel === 'apple-touch-icon-precomposed')
+      .filter(el => {
```

Comment:
.filter(el => !!el.href)

Reply:
then el.href is `string | null` and would need to be `el.href || ''` to make ts be quiet.  How do I make ts love me?
```javascript
// Check that the href is valid
try {
    new URL(el.href, artifacts.URL.finalUrl);
```

Reply:
OHHHHHHHH got it.